gpii/node_modules/flowManager/test/BrowserChannelTests.js

```diff
@@ -0,0 +1,400 @@
+        "highContrastEnabled": true,
+        "invertColours": false,
+        "magnifierEnabled": true,
+        "magnification": 2,
+        "fontSize": "medium",
+        "simplifier": false,
+        "highContrastTheme": "black-white"
+    }
+};
+
+gpii.tests.flowManager.browserChannel.testLoginResponseChrome = function (data) {
+    jqUnit.assertEquals("Response is correct", "User with token " +
+        gpii.tests.flowManager.browserChannel.chromeUserToken + " was successfully logged in.", data);
+};
+
+gpii.tests.flowManager.browserChannel.testLogoutResponseChrome = function (data) {
+    jqUnit.assertEquals("Response is correct", "User with token " +
+        gpii.tests.flowManager.browserChannel.chromeUserToken + " was successfully logged out.", data);
+};
+
+gpii.tests.flowManager.browserChannel.testLoginResponseFirefox = function (data) {
+    jqUnit.assertEquals("Response is correct", "User with token " +
+        gpii.tests.flowManager.browserChannel.firefoxUserToken + " was successfully logged in.", data);
+};
+
+gpii.tests.flowManager.browserChannel.testLogoutResponseFirefox = function (data) {
+    jqUnit.assertEquals("Response is correct", "User with token " +
+        gpii.tests.flowManager.browserChannel.firefoxUserToken + " was successfully logged out.", data);
+};
+
+
+gpii.tests.flowManager.browserChannel.checkConnectionRequest = function (data, client) {
+    jqUnit.assertEquals("Connection suceeded", "A request to browserChannel has been successfully processed", data);
+
+    // Let's connect to 'onBrowserSettingsChanged' in order to keep track of the changes
+    // in the settings during the tests.
+    //
+    client.socket.on("onBrowserSettingsChanged", client.events.onBrowserSettingsChanged.fire);
+};
+
+gpii.tests.flowManager.browserChannel.checkFirstClient = function () {
+    jqUnit.assertEquals("'org.chrome.cloud4chrome' is registered as a client",
+        "org.chrome.cloud4chrome",
+        Object.keys(gpii.settingsHandlers.WebSockets.clients)
+    );
+};
+
+gpii.tests.flowManager.browserChannel.checkSettingsForChrome = function () {
+    jqUnit.assertEquals("The settings for 'org.chrome.cloud4chrome' are the expected",
+        JSON.stringify(gpii.tests.flowManager.browserChannel.payloads.expectedChromeSettings),
+        JSON.stringify(gpii.settingsHandlers.WebSockets.getSettingsForId("org.chrome.cloud4chrome"))
+    );
+};
+
+gpii.tests.flowManager.browserChannel.checkSettingsForFirefox = function () {
+    jqUnit.assertEquals("The settings for 'org.mozilla.cloud4firefox' are the expected",
+        JSON.stringify(gpii.tests.flowManager.browserChannel.payloads.expectedFirefoxSettings),
+        JSON.stringify(gpii.settingsHandlers.WebSockets.getSettingsForId("org.mozilla.cloud4firefox"))
+    );
+};
+
+
+gpii.tests.flowManager.browserChannel.checkNoClients = function () {
+    jqUnit.assertEquals("No clients are registered yet", "", Object.keys(gpii.settingsHandlers.WebSockets.clients));
```

Comment:
I don't see why this should work - shouldn't the value be [] rather than "" ? Looks like it's relying on a misbehaviour of QUnit. Better to use jqUnit.assertDeepEq together with []